### PR TITLE
chore: Reorganize the installation config settings

### PR DESCRIPTION
**Pull request by sojan-official**
- Reorganizing installation config settings to move more configurations into UI from environment variables
- Changes to installation config to support premium plans in enterprise edition

### CODE REVIEW

File: app/jobs/internal/check_new_versions_job.rb [hunk 1]
```diff
@@ -4,21 +4,17 @@ class Internal::CheckNewVersionsJob < ApplicationJob
...
-
-    ::Redis::Alfred.set(::Redis::Alfred::LATEST_CHATWOOT_VERSION, instance_info['version'])
-    update_installation_config(key: 'INSTALLATION_PRICING_PLAN', value: instance_info['plan'])
-    update_installation_config(key: 'INSTALLATION_PRICING_PLAN_QUANTITY', value: instance_info['plan_quantity'])
-    update_installation_config(key: 'CHATWOOT_SUPPORT_WEBSITE_TOKEN', value: instance_info['chatwoot_support_website_token'])
-    update_installation_config(key: 'CHATWOOT_SUPPORT_IDENTIFIER_HASH', value: instance_info['chatwoot_support_identifier_hash'])
-    update_installation_config(key: 'CHATWOOT_SUPPORT_SCRIPT_URL', value: instance_info['chatwoot_support_script_url'])
+    @instance_info = ChatwootHub.sync_with_hub
+    update_version_info
   end
```
Comment (sojan-official) [inline app/jobs/internal/check_new_versions_job.rb:9]: moving this to Enterprise code

File: app/controllers/super_admin/app_configs_controller.rb [hunk 2]
```diff
@@ -22,19 +22,24 @@ def create
...
 
   def allowed_configs
-    @allowed_configs = %w[FB_APP_ID FB_VERIFY_TOKEN FB_APP_SECRET]
+    @allowed_configs = case @config
+                       when 'facebook'
+                         %w[FB_APP_ID FB_VERIFY_TOKEN FB_APP_SECRET IG_VERIFY_TOKEN ENABLE_MESSENGER_CHANNEL_HUMAN_AGENT]
+                       when 'email'
+                         ['MAILER_INBOUND_EMAIL_DOMAIN']
+                       else
+                         %w[ENABLE_ACCOUNT_SIGNUP INSTALLATION_EVENTS_WEBHOOK_URL]
```
Comment (pranavrajs) [inline app/controllers/super_admin/app_configs_controller.rb:41]:
```suggestion
                         %w[ENABLE_ACCOUNT_SIGNUP]
```

INSTALLATION_EVENTS_WEBHOOK_URL is only used for internal use cases. I would suggest to avoid exposing those.

File: config/installation_config.yml [hunk 3]
```diff
@@ -41,67 +56,138 @@
...
-  value:
+# ------- End of Branding Related Config ------- #
+
+
+
+# ------- Signup & Account Related Config ------- #
+- name: ENABLE_ACCOUNT_SIGNUP
+  display_title: 'Enable Account Signup'
+  type: boolean
+  description: 'Allow users to signup for new accounts'
```
Comment (pranavrajs) [inline config/installation_config.yml:67]:
```suggestion
```

This seems to be a duplicate of the line below.

File: config/installation_config.yml [hunk 4]
```diff
@@ -41,67 +56,138 @@
...
+  value: false
+  description: 'Allow users to signup for new accounts'
   locked: false
 - name: CREATE_NEW_ACCOUNT_FROM_DASHBOARD
   value: false
+  description: 'Allow users to create new accounts from the dashboard'
```
Comment (pranavrajs) [inline config/installation_config.yml:73]: Let us hide this if possible.

File: config/installation_config.yml [hunk 5]
```diff
@@ -41,67 +56,138 @@
...
   locked: false
-- name: INSTALLATION_EVENTS_WEBHOOK_URL
+- name: HCAPTCHA_SITE_KEY
```
Comment (pranavrajs) [inline config/installation_config.yml:75]: Does changing the order affect us any way?

File: config/installation_config.yml [hunk 6]
```diff
@@ -41,67 +56,138 @@
...
   locked: false
-- name: CHATWOOT_INBOX_TOKEN
+- name: HCAPTCHA_SERVER_KEY
   value:
   locked: false
-- name: CHATWOOT_INBOX_HMAC_KEY
+- name: INSTALLATION_EVENTS_WEBHOOK_URL
   value:
+  display_title: 'System events Webhook URL'
+  description: 'The URL to which the system events like new accounts created will be sent'
```
Comment (pranavrajs) [inline config/installation_config.yml:84]: Same comments as above. Let us not expose this to the end user. This is only used internally.

File: config/installation_config.yml [hunk 7]
```diff
@@ -41,67 +56,138 @@
...
   value: false
+  description: 'Enable direct uploads to cloud storage'
   locked: false
-- name: HCAPTCHA_SITE_KEY
+# ------- End of Account Related Config ------- #
+
+
+
+# ------- Email Related Config ------- #
+- name: MAILER_INBOUND_EMAIL_DOMAIN
```
Comment (pranavrajs) [inline config/installation_config.yml:96]: We could potentially add a title field explaining the variable in plain English. That can be an enhancement.

File: config/installation_config.yml [hunk 8]
```diff
@@ -41,67 +56,138 @@
...
+
+
+# ------- Facebook Channel Related Config ------- #
+- name: FB_APP_ID 
+  display_title: 'Facebook App ID'
   locked: false
-- name: DISABLE_USER_PROFILE_UPDATE
-  value: false
+- name: FB_VERIFY_TOKEN 
+  display_title: 'Facebook Verify Token'
```
Comment (pranavrajs) [inline config/installation_config.yml:111]:
```suggestion
  display_title: 'Facebook webhook verification token'
```

File: config/installation_config.yml [hunk 9]
```diff
@@ -41,67 +56,138 @@
...
+  locked: false
+- name: FB_APP_SECRET
+  display_title: 'Facebook App Secret'
+  locked: false
+- name: IG_VERIFY_TOKEN
+  display_title: 'Instagram Verify Token'
```
Comment (pranavrajs) [inline config/installation_config.yml:117]:
```suggestion
  display_title: 'Instagram webhook verification token'
```

File: config/installation_config.yml [hunk 10]
```diff
@@ -41,67 +56,138 @@
...
 - name: CHATWOOT_CLOUD_PLANS
   value:
+  description: 'Config to store stripe plans for cloud'
 - name: DEPLOYMENT_ENV
   value: self-hosted
-- name: CSML_EDITOR_HOST
+  description: 'The deployment environment of the installation, to differentiate between Chatwoot cloud and self-hosted'
+- name: ANALYTICS_TOKEN
   value:
+  description: 'The june analytics token for Chatwoot cloud'
```
Comment (pranavrajs) [inline config/installation_config.yml:144]:
```suggestion
  description: 'The June.so analytics token for Chatwoot cloud'
```

File: config/installation_config.yml [hunk 11]
```diff
@@ -41,67 +56,138 @@
...
+# ------- Chatwoot Internal Config for Self Hosted ----#
 - name: INSTALLATION_PRICING_PLAN
   value: 'community'
+  description: 'The pricing plan for the installation, retrieved from the billing API'
 - name: INSTALLATION_PRICING_PLAN_QUANTITY
   value: 0
+  description: 'The number of licenses purchased for the installation, retrieved from the billing API'
 - name: CHATWOOT_SUPPORT_WEBSITE_TOKEN
   value:
+  description: 'The Chatwoot Support Website Token for Contact Support in Self Hosted'
```
Comment (pranavrajs) [inline config/installation_config.yml:157]:
```suggestion
  description: 'The Chatwoot support website token for contacting support from self-hosted installations'
```

File: config/installation_config.yml [hunk 12]
```diff
@@ -41,67 +56,138 @@
...
+# ------- End of Chatwoot Internal Config for Cloud ----#
+
+
+# ------- Chatwoot Internal Config for Self Hosted ----#
 - name: INSTALLATION_PRICING_PLAN
   value: 'community'
+  description: 'The pricing plan for the installation, retrieved from the billing API'
 - name: INSTALLATION_PRICING_PLAN_QUANTITY
   value: 0
+  description: 'The number of licenses purchased for the installation, retrieved from the billing API'
```
Comment (pranavrajs) [inline config/installation_config.yml:154]:
```suggestion
  description: 'The number of licenses purchased for the installation, retrieved from the billing API'
```

File: config/installation_config.yml [hunk 13]
```diff
@@ -41,67 +56,138 @@
...
 - name: CHATWOOT_SUPPORT_WEBSITE_TOKEN
   value:
+  description: 'The Chatwoot Support Website Token for Contact Support in Self Hosted'
 - name: CHATWOOT_SUPPORT_SCRIPT_URL
   value:
+  description: 'The Chatwoot Support Script URL for Contact Support in Self Hosted'
```
Comment (pranavrajs) [inline config/installation_config.yml:160]:
```suggestion
  description: 'The Chatwoot script base URL, to display the "Contact Support" option on the billing page'
```

File: config/installation_config.yml [hunk 14]
```diff
@@ -41,67 +56,138 @@
...
 - name: CHATWOOT_SUPPORT_IDENTIFIER_HASH
   value:
+  description: 'The Chatwoot Support Identifier Hash for Contact Support in Self Hosted'
```
Comment (pranavrajs) [inline config/installation_config.yml:163]:
```suggestion
  description: 'The Chatwoot identifier hash, to validate the contact in the live chat window.'
```

File: config/installation_config.yml [hunk 15]
```diff
@@ -41,67 +56,138 @@
...
+  value:
+  description: 'Custom thumbnail for the API channel'
+- name: LOGOUT_REDIRECT_LINK
+  value: /app/login
+  locked: false
+  description: 'Redirect to a different link after logout'
+- name: DISABLE_USER_PROFILE_UPDATE
+  value: false
+  locked: false
+  description: 'Disabled rendering profile update page for users'
```
Comment (pranavrajs) [inline config/installation_config.yml:180]:
```suggestion
  description: 'Disable rendering profile update page for users'
```

File: config/installation_config.yml [hunk 16]
```diff
@@ -41,67 +56,138 @@
...
+# Moves everything below this line to appropriate section with descriptions #
+# -------Misc Config ------- #
+- name: CSML_BOT_HOST
+  value:
+  locked: false
+- name: CSML_BOT_API_KEY
+  value:
+  locked: false
+- name: CSML_EDITOR_HOST
+  value:
```
Comment (pranavrajs) [inline config/installation_config.yml:193]:
```suggestion
```

Let us remove this as CSML is discontinued.

**Review by pranavrajs — APPROVED** [review]

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Some minor comments, otherwise looks fine.

File: config/installation_config.yml (hunk 5 above)
Comment (sojan-official) [reply config/installation_config.yml:75]: Not in the case of installation config. we take this as array of objects and process accordingly.

File: config/installation_config.yml (hunk 6 above)
Comment (sojan-official) [reply config/installation_config.yml:84]: Added description here so that we have a cue of what this config is used for. removed showing this from the UI

File: config/installation_config.yml (hunk 8 above)
Comment (sojan-official) [reply config/installation_config.yml:111]: long name will break the UI. Added this as description instead

File: config/installation_config.yml (hunk 9 above)
Comment (sojan-official) [reply config/installation_config.yml:117]: same comment as verify token
